lib/src/main/kotlin/WeatherSensorService/Internal/MockWeatherSensorReader.kt

```diff
@@ -57,11 +61,15 @@ class MockWeatherSensorReader: WeatherSensorReaderType {
      private fun reportSensorReadings() {
          flow.tryEmit(
              WeatherSensorReading(
-                Random.nextDouble(-40.0, 40.0),
-                Random.nextDouble(0.0,100.0),
-                Random.nextDouble(95.0,105.0),
+                GenerateTemperatureReadings((waveLocation/100.0),Constants.TEMPERATURE_RANGE),
+                GenerateTemperatureReadings((waveLocation/100.0),Constants.HUMIDITY_RANGE),
+                GenerateTemperatureReadings((waveLocation/100.0),Constants.PRESSURE_RANGE),
```

Comment:
- The percentage/amplitude is calculated repeatedly 3 times `waveLocation/100.0` (should be done once via a variable).
- The parenthesis around `waveLocation/100.0` are unnecessary
- There should be a space after the comma between the parameters.
- Lines 64-66 are only indented 3 spaces

So if you made the top-level function an extension of `ClosedRange<Double>`, then this would look like:
```
WeatherSensorReading(
    Constants.TEMPERATURE_RANGE.calculateSineWave(amplitude),
    ...
)
```